internal/characters/arlecchino/attack.go

```diff
@@ -0,0 +1,204 @@
+)
+
+var (
+	attackFrames          [][]int
+	attackHitmarks        = [][]int{{11}, {16}, {17}, {24, 35}, {21}, {44}}
+	attackHitlagHaltFrame = [][]float64{{0.02}, {0.02}, {0.02}, {0, 0}, {0}, {0.02}}
+	attackDefHalt         = [][]bool{{true}, {true}, {true}, {true, false}, {false}, {true}}
+	attackStrikeTypes     = [][]attacks.StrikeType{
+		{attacks.StrikeTypeSlash},
+		{attacks.StrikeTypeSlash},
+		{attacks.StrikeTypeSlash},
+		{attacks.StrikeTypeSlash, attacks.StrikeTypeSlash},
+		{attacks.StrikeTypeSpear},
+		{attacks.StrikeTypeSlash},
+	}
+	attackHitboxes = [][][][]float64{
+		{
+			{{1.9, 3}},     // box
+			{{2.6}},        // fan
+			{{1.9, 4}},     // box
+			{{2.8}, {2.8}}, // circle, circle
+			{{2.5}},        // circle
+			{{3}},          // circle
+		},
+		{
+			{{1.9, 4.2}},   // box
+			{{3.1}},        // fan
+			{{3.4, 5.6}},   // box
+			{{3.3}, {3.3}}, // circle, circle
+			{{2.8}},        // circle
+			{{3.7}},        // circle
+		},
+	}
+	attackOffsets = [][][]float64{
+		{{0, -0.15}},
+		{{0, 0.5}},
+		{{0, -1.2}},
+		{{-0.5, 0.7}, {-0.5, 0.7}},
+		{{0, 2.4}},
+		{{0, 2.5}},
+	}
+
+	attackFanAngles = [][]float64{{360}, {300}, {360}, {360, 360}, {360}, {360}}
+)
+
+const naBuffKey = "masque-of-the-red-death"
+const bondConsumeICDKey = "bond-consume-icd"
+const normalHitNum = 6
+
+func init() {
+	attackFrames = make([][]int, normalHitNum)
+
+	attackFrames[0] = frames.InitNormalCancelSlice(attackHitmarks[0][0], 24)
+	attackFrames[0][action.ActionAttack] = 11
+	attackFrames[0][action.ActionCharge] = 17
+
+	attackFrames[1] = frames.InitNormalCancelSlice(attackHitmarks[1][0], 31)
+	attackFrames[1][action.ActionAttack] = 21
+	attackFrames[1][action.ActionCharge] = 26
+
+	attackFrames[2] = frames.InitNormalCancelSlice(attackHitmarks[2][0], 39)
+	attackFrames[2][action.ActionAttack] = 31
+	attackFrames[2][action.ActionCharge] = 33
+
+	attackFrames[3] = frames.InitNormalCancelSlice(attackHitmarks[3][1], 55)
+	attackFrames[3][action.ActionAttack] = 49
+	attackFrames[3][action.ActionCharge] = 49
+
+	attackFrames[4] = frames.InitNormalCancelSlice(attackHitmarks[4][0], 43)
+	attackFrames[4][action.ActionAttack] = 30
+	attackFrames[4][action.ActionCharge] = 33
+
+	attackFrames[5] = frames.InitNormalCancelSlice(attackHitmarks[5][0], 59)
+	attackFrames[5][action.ActionAttack] = 58
+	attackFrames[5][action.ActionCharge] = 55
+}
+
+func (c *char) naBuff() {
+	c.Core.Events.Subscribe(event.OnHPDebt, func(args ...interface{}) bool {
+		target := args[0].(int)
+		if target != c.Index {
+			return false
+		}
+		// TODO: Remove when BoL changes get logged for all characters
+		c.Core.Log.NewEvent("Bond of Life changed", glog.LogCharacterEvent, c.Index).
+			Write("arle_hp_debt", c.CurrentHPDebt()).
+			Write("arle_hp_debt%", c.CurrentHPDebt()/c.MaxHP())
+		if c.CurrentHPDebt() >= c.MaxHP()*0.3 {
+			// can't use negative duration or else `if .arlecchino.status.masque-of-the-red-death` won't work
+			c.AddStatus(naBuffKey, 999999, false)
```

Comment:
you can use SetTag for it

Reply:
Do I do 

```
c.AddStatus(naBuffKey, -1, false)
c.SetTag(naBuffKey, 1)
```
Then? I'd like the status to appear in the Sample and be accessible using `.arlecchino.status.masque-of-the-red-death`

Reply:
I tried setting the tag, but the tag only works with `.arlecchino.tags.masque-of-the-red-death` and not `.arlecchino.status.masque-of-the-red-death` or `.arlecchino.mods.masque-of-the-red-death`

Reply:
I think I will just do `c.AddStatus(naBuffKey, -1, false)` and expect users to check her buff status by checking her bolratio. `.arlecchino.bolratio >= 0.3`

Reply:
> I tried setting the tag, but the tag only works with .arlecchino.tags.masque-of-the-red-death and not .arlecchino.status.masque-of-the-red-death or .arlecchino.mods.masque-of-the-red-death

is that really necessary to use the status and not the tag?

Reply:
Statuses show up in the Sample, but tags don't. So I prioritized the status appearing in the sample, even if the status conditional doesn't work.